From 4a166a50f6a1b50e602d5218e9e0fa1d3ab917dd Mon Sep 17 00:00:00 2001
From: Arie Brown <abrown@habana.ai>
Date: Thu, 27 Aug 2020 17:18:11 +0300
Subject: [PATCH] [SW-19146] fit: add creation fit per device

for supporting creating dsv fit file only for goya,
add the ability to create a fit by a dedicate function for a device.

Change-Id: If2544e3e2b69d1e5c84cc44da5d57e461e7b2491
Signed-off-by: Arie Brown <abrown@habana.ai>
---

diff --git a/board/habanalabs/common/post-image-fit.sh b/board/habanalabs/common/post-image-fit.sh
index 3ca1f21..293ba84 100755
--- a/board/habanalabs/common/post-image-fit.sh
+++ b/board/habanalabs/common/post-image-fit.sh
@@ -31,6 +31,7 @@
 HL_SEC_KEY_INDEX=`grep HL_SEC_KEY_INDEX $HL_SEC_FILE | sed 's/^.*\(0x.*\)$/\1/'`
 HL_SEC_FW_VERSION=`grep HL_SEC_FW_VERSION $HL_SEC_FILE | awk '{ print $NF }'`
 
+HL_DEV_BASE=${BR2_EXTERNAL_HABANALABS_PATH}/board/habanalabs/${HL_DEV}
 COMMON_BASE=${BR2_EXTERNAL_HABANALABS_PATH}/board/habanalabs/common
 UBOOT_ITS_FILE=uboot-${HL_DEV}_${HL_TARGET}.its
 UBOOT_ITB_FILE=uboot-fit.itb
@@ -60,6 +61,15 @@
 KEYS_DIR=${BR2_EXTERNAL_HABANALABS_PATH}/board/habanalabs/common/key/u-boot/
 ITS_SED_PARAMS+="s#@KEY_DTB@#${KEYS_DIR}/hl_rsa3072.dtb#g;"
 
+function generate_device_fit 
+{
+	return
+}
+
+if [[ -f ${HL_DEV_BASE}/post-image-fit.sh ]]; then
+	source ${HL_DEV_BASE}/post-image-fit.sh
+fi
+
 pushd ${BINARIES_DIR}
 	if [ ${HL_TARGET} == "onix" ]; then
 		sed ${ITS_SED_PARAMS} ${SPI_FPGA_ITS_TEMPLATE} > ${SPI_ITS_FILE}
@@ -78,4 +88,6 @@
 	${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${LINUX_ITS_FILE} -k ${KEYS_DIR} ${LINUX_ITB_FILE}
 	# KMD like this to be 4B aligned in size
 	dd status=none conv=sync bs=4 if=${LINUX_ITB_FILE} > ${HL_DEV}-fit.itb
+
+	generate_device_fit
 popd
